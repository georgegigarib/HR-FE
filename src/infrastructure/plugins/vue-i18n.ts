import { en } from '@/infrastructure/local/en.local'
import { es } from '@/infrastructure/local/es.local'

import { createI18n } from 'vue-i18n'

const [userLanguage] = navigator.language.split('-')

const vueI18n = createI18n({
  legacy: false,
  locale: userLanguage,
  fallbackLocale: 'en',
  missing: (locale, key) => {
    console.warn(`Missing translation key: ${key}, in locale: ${locale}`)
    return ''
  },
  messages: {
    en,
    es
  }
})

export { vueI18n }
