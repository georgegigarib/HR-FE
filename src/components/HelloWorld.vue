<template>
  <div class="greetings">
    <h1>{{ t('hello') }}</h1>
    <h1 class="green">{{ msg }}</h1>
    <h3>
      Youâ€™ve successfully created a project with
      <a href="https://vitejs.dev/" target="_blank" rel="noopener">Vite</a> +
      <a href="https://vuejs.org/" target="_blank" rel="noopener">Vue 3</a>. What's next?
    </h3>
    <button data-test="button-for-toast" @click="showToast()" class="bg-white">Show Toast</button>
  </div>
</template>

<script setup lang="ts">
import { useI18n } from 'vue-i18n'
import { toast } from 'vue3-toastify'

const { t } = useI18n()
defineProps<{ msg: string }>()

const showToast = () => {
  toast.success('Hello, Vite + Vue 3!')
}
</script>

<style scoped>
h1 {
  font-weight: 500;
  font-size: 2.6rem;
  position: relative;
  top: -10px;
}

h3 {
  font-size: 1.2rem;
}

.greetings h1,
.greetings h3 {
  text-align: center;
}

@media (min-width: 1024px) {
  .greetings h1,
  .greetings h3 {
    text-align: left;
  }
}
</style>
