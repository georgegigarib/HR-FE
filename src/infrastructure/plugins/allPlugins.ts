import { vueI18n } from '@/infrastructure/plugins/vue-i18n'

export { vueI18n }

export const plugins = [vueI18n]
