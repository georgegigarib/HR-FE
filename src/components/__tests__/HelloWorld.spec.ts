import { mount, VueWrapper } from '@vue/test-utils'
import { describe, expect, it, beforeEach } from 'vitest'
import { plugins } from '../../infrastructure/plugins/allPlugins'
import HelloWorld from '../HelloWorld.vue'

describe('HelloWorld', () => {
	let wrapper: VueWrapper

	beforeEach(() => {
		wrapper = mount(HelloWorld, {
			props: { msg: 'Hello Vitest' },
			global: {
				plugins: [...plugins],
			},
		})
	})

	it('renders properly', () => {
		expect(wrapper.text()).toContain('Hello Vitest')
	})

  it('has expected html structure', (): void => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
