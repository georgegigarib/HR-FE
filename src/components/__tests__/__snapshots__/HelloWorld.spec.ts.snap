// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HelloWorld > has expected html structure 1`] = `
<div
  class="greetings"
  data-v-e17ea971=""
>
  <h1
    data-v-e17ea971=""
  >
    hellow World this is george
  </h1>
  <h1
    class="green"
    data-v-e17ea971=""
  >
    Hello Vitest
  </h1>
  <h3
    data-v-e17ea971=""
  >
     Youâ€™ve successfully created a project with 
    <a
      data-v-e17ea971=""
      href="https://vitejs.dev/"
      rel="noopener"
      target="_blank"
    >
      Vite
    </a>
     + 
    <a
      data-v-e17ea971=""
      href="https://vuejs.org/"
      rel="noopener"
      target="_blank"
    >
      Vue 3
    </a>
    . What's next? 
  </h3>
  <button
    class="bg-white"
    data-test="button-for-toast"
    data-v-e17ea971=""
  >
    Show Toast
  </button>
</div>
`;
